tests/test_compound_comparison_levels.py

```diff
@@ -0,0 +1,123 @@
+import pandas as pd
+
+from splink.duckdb.duckdb_linker import DuckDBLinker
+import splink.duckdb.duckdb_comparison_library as cl
+import splink.duckdb.duckdb_comparison_level_library as cll
+
+
+def test_compound_comparison_level():
+
+    df = pd.DataFrame(
+        [
+            {
+                "unique_id": 1,
+                "first_name": "John",
+                "middle_name": "James",
+                "surname": "Smith",
+                "city": "Brighton",
+            },
+            {
+                "unique_id": 2,
+                "first_name": "Mary",
+                "middle_name": "Harriet",
+                "surname": "Jones",
+                "city": "Brighton",
+            },
+            {
+                "unique_id": 3,
+                "first_name": "Jane",
+                "middle_name": "Joan",
+                "surname": "Taylor",
+                "city": "Brighton",
+            },
+            {
+                "unique_id": 4,
+                "first_name": "John",
+                "middle_name": "Blake",
+                "surname": "Jones",
+                "city": "Brighton",
+            },
+            {
+                "unique_id": 5,
+                "first_name": "Jane",
+                "middle_name": "Joan",
+                "surname": "Taylor",
+                "city": "Brighton",
+            },
+            {
+                "unique_id": 6,
+                "first_name": "Gill",
+                "middle_name": "Harriet",
+                "surname": "Greene",
+                "city": "Brighton",
+            },
+            {
+                "unique_id": 7,
+                "first_name": "Owen",
+                "middle_name": "James",
+                "surname": "Smith",
+                "city": "Brighton",
+            },
+            {
+                "unique_id": 8,
+                "first_name": "Sarah",
+                "middle_name": "Simone",
+                "surname": "Williams",
+                "city": "Brighton",
+            },
+        ]
+    )
+
+    def col_is_match(col):
+        return f"({col}_l = {col}_r)"
+
+    def col_is_null(col):
+        return f"({col}_l IS NULL OR {col}_r IS NULL)"
+
+    sql_two_out_of_three_match = (
```

Comment:
Rename this variable to `sql_ands_joined_with_ors` or `sql_non_CNF` or something to make it more clear what this is testing.